### PR TITLE
RSDK-2925 - Add MoveOnGlobe and message additions to Common and Motion APIs

### PR DESCRIPTION
**Pull request by nfranczak**
new PR created to move geo_obstacle.go from the referenceframe package into the spatialmath package

old PR here
https://github.com/viamrobotics/rdk/pull/2330

### CODE REVIEW

**Review by biotinker — APPROVED** [review]
LGTM

File: services/motion/client.go [hunk 1]
```diff
@@ -97,6 +101,66 @@ func (c *client) MoveOnMap(
...
+		Destination:        &commonpb.GeoPoint{Latitude: destination.Lat(), Longitude: destination.Lng()},
+		MovementSensorName: protoutils.ResourceNameToProto(movementSensorName),
+		Extra:              ext,
+	}
+
+	// Optionals
+	if !math.IsNaN(heading) {
+		req.Heading = &heading
+	}
+	if len(obstacles) > 0 {
```
Comment (raybjork) [inline services/motion/client.go:136]: I don't think the if statement is necessary here

File: services/motion/client.go [hunk 2]
```diff
@@ -97,6 +101,66 @@ func (c *client) MoveOnMap(
...
+		for _, eachObst := range obstacles {
+			convObst, err := spatialmath.GeoObstacleToProtobuf(eachObst)
+			if err != nil {
+				return false, err
+			}
+			obstaclesProto = append(obstaclesProto, convObst)
+		}
+		req.Obstacles = obstaclesProto
+	}
+	reqLinear := float32(linearVelocity)
```
Comment (raybjork) [inline services/motion/client.go:147]: why are these being cast to `float32`

**Review by raybjork — CHANGES_REQUESTED** [review]

File: services/motion/server.go [hunk 3]
```diff
@@ -61,6 +63,52 @@ func (server *serviceServer) MoveOnMap(ctx context.Context, req *pb.MoveOnMapReq
...
+	for _, eachProtoObst := range obstaclesProto {
+		convObst, err := spatialmath.GeoObstacleFromProtobuf(eachProtoObst)
+		if err != nil {
+			return nil, err
+		}
+		obstacles = append(obstacles, convObst)
+	}
+	linear := math.NaN()
+	if req.LinearMetersPerSec != nil {
+		linear = float64(req.GetLinearMetersPerSec())
```
[the remark this answers is not in the document]
Comment (biotinker) [reply services/motion/server.go:91]: That's a float32 in the API

File: services/motion/client.go (hunk 2 above)
Comment (biotinker) [reply services/motion/client.go:147]: Related to above; the API type for these is `float` (float32) not `double` (float64)

File: services/motion/client.go (hunk 1 above)
Comment (biotinker) [reply services/motion/client.go:136]: It's mildly more efficient to have it in

File: services/motion/client.go (hunk 2 above)
Comment (raybjork) [reply services/motion/client.go:147]: Interesting.  Thanks

File: services/motion/server.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (nfranczak) [reply services/motion/server.go:91]: i think we need to. looking inside `motion.pb.go` we have this definition where `LinearMetersPerSec *float32 `
```
type MoveOnGlobeRequest struct {
	state         protoimpl.MessageState
	sizeCache     protoimpl.SizeCache
	unknownFields protoimpl.UnknownFields

	// Name of the motion service
	Name string `protobuf:"bytes,1,opt,name=name,proto3" json:"name,omitempty"`
	// Destination, encoded as a GeoPoint
	Destination *v1.GeoPoint `protobuf:"bytes,2,opt,name=destination,proto3" json:"destination,omitempty"`
	// Optional compass heading to achieve at the destination, in degrees [0-360)
	Heading *float64 `protobuf:"fixed64,3,opt,name=heading,proto3,oneof" json:"heading,omitempty"`
	// Component on the robot to move to the specified destination
	ComponentName *v1.ResourceName `protobuf:"bytes,4,opt,name=component_name,json=componentName,proto3" json:"component_name,omitempty"`
	// Name of the movement sensor which will be used to check robot location
	MovementSensorName *v1.ResourceName `protobuf:"bytes,5,opt,name=movement_sensor_name,json=movementSensorName,proto3" json:"movement_sensor_name,omitempty"`
	// Obstacles to be considered for motion planning
	Obstacles []*v1.GeoObstacle `protobuf:"bytes,6,rep,name=obstacles,proto3" json:"obstacles,omitempty"`
	// Optional linear velocity to target when moving
	LinearMetersPerSec *float32 `protobuf:"fixed32,7,opt,name=linear_meters_per_sec,json=linearMetersPerSec,proto3,oneof" json:"linear_meters_per_sec,omitempty"`
	// Optional angular velocity to target when turning
	AngularDegPerSec *float32 `protobuf:"fixed32,8,opt,name=angular_deg_per_sec,json=angularDegPerSec,proto3,oneof" json:"angular_deg_per_sec,omitempty"`
	// Additional arguments to the method
	Extra *structpb.Struct `protobuf:"bytes,99,opt,name=extra,proto3" json:"extra,omitempty"`
}
```

**Review by raybjork — APPROVED** [review]
Just the comment I left but its super minor so LGTM

**Pull request merged**